src/Sema.zig

```diff
@@ -26327,8 +26327,10 @@ fn zirMemcpy(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!void
     try sema.validateRuntimeValue(block, dest_src, dest_ptr);
     try sema.validateRuntimeValue(block, src_src, src_ptr);
 
+    const dest_has_bits = dest_elem_ty.hasRuntimeBits(zcu);
     // Aliasing safety check.
-    if (block.wantSafety()) {
+    // Only emit check if there's any copying to be done
+    if (block.wantSafety() and dest_has_bits) {
```

Comment:
Would it be better to just do an early exit if there are no runtime bits? I think such an early exit could also be made earlier, maybe after the len checks on line 26182.

Reply:
yeah that would be cleaner

Reply:
perhaps afer line 26290 right after getting the destination element type?

Reply:
thank you for your feedback!